### PR TITLE
VZ-10404 module upgrade pre post controller sync

### PR DESCRIPTION
**Pull request by pfmackin**
This is Ben's branch with unit tests fixes.
From Ben:
This change creates global atomics that synchronize the new experimental verrazzano controller and the legacy verrazzano controller to facilitate module install and upgrade.

### CODE REVIEW

File: platform-operator/controllers/verrazzano/reconcile/status.go [hunk 1]
```diff
@@ -366,3 +369,35 @@ func (r *Reconciler) setUninstallCondition(log vzlog.VerrazzanoLogger, vz *insta
 	}
 	return r.updateStatus(log, vz, msg, newCondition, nil)
 }
+
+func (r *Reconciler) modulesReady(ctx spi.ComponentContext) (bool, error) {
```
Comment (michael-cico) [inline platform-operator/controllers/verrazzano/reconcile/status.go:373]: even though it's (hopefully) temporary, maybe add a comment here about what it's doing.  Next pass.

File: platform-operator/experimental/catalog/catalog.yaml [hunk 2]
```diff
@@ -35,9 +35,9 @@ modules:
   - name: verrazzano-application-operator
     version: 2.0.0
   - name: weblogic-operator
-    version: 4.1.0
+    version: 4.1.0-1
```
Comment (michael-cico) [inline platform-operator/experimental/catalog/catalog.yaml:38]: why are we bumping these here?

File: platform-operator/experimental/controllers/module/component-handler/migration/migration.go [hunk 3]
```diff
@@ -62,5 +62,5 @@ func (h migrationHandler) UpdateStatusIfAlreadyInstalled(ctx handlerspi.HandlerC
 	}
 
 	// Set the module status condition, installed generation and installed version
-	return modulestatus.UpdateModuleStatusToInstalled(ctx, module, vzcr.Status.Version, compStatus.LastReconciledGeneration)
+	return modulestatus.UpdateModuleStatusToInstalled(ctx, module, vzcr.Status.Version, 0)
```
Comment (michael-cico) [inline platform-operator/experimental/controllers/module/component-handler/migration/migration.go:65]: So we want to force a reconcile here?

**Review by michael-cico — DISMISSED** [review]

File: platform-operator/controllers/verrazzano/reconcile/status.go (hunk 1 above)
Comment (pfmackin) [reply platform-operator/controllers/verrazzano/reconcile/status.go:374]: fixed

File: platform-operator/experimental/controllers/module/component-handler/migration/migration.go (hunk 3 above)
Comment (pfmackin) [reply platform-operator/experimental/controllers/module/component-handler/migration/migration.go]: fixed

**Review by michael-cico — APPROVED** [review]

**Pull request merged**